src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/Whitelist.java

```diff
@@ -66,6 +66,70 @@
 
     public abstract boolean permitsStaticFieldSet(@Nonnull Field field, @CheckForNull Object value);
 
+    // Utility methods for creating canonical string representations of the signature
+    public static final StringBuilder joinWithSpaces(StringBuilder b, String[] types) {
```

Comment:
`joinWithSpaces` and `getName` (both variants) and `argumentTypes` are all just relocated from EnumeratingWhitelist unchanged (allowing them to be reused here).